### ISSUE TITLE
One-way audio due to OPUS payload being misinterpreted as DTMF

### ISSUE THREAD

**Issue by olofssonanton** [signalwire/freeswitch#1965]
### Describe the bug
When bridging a call (sofia to sofia) to a terminal _not_ using `183 Session Progress`, the B-side's **telephone-event** is erroneously setup to be received using the payload type intended for **OPUS** (as stated in both local and remote SDP), resulting in the incoming audio stream from B being misinterpreted as DTMF by the FreeSWITCH.

### To Reproduce
This should be reproducible in a vanilla configuration, using similar SDP/payload types as shown in the attached logs, in something like sipp.
#### To succeed:
1. Bridge call from A to B.
2. B responds with `183 Session Progress` before `200 OK` (using the same SDP and payload types).
3. DTMF send/receive payload types can be confirmed to be correct in the log `Set 2833 dtmf send payload to 102 recv payload to 102`.
4. No audio issues.

#### To fail:
1. Bridge call from A to B.
2. B responds with `200 OK`.
3. DTMF send/receive payload types can be confirmed to be incorrect in the log `Set 2833 dtmf send payload to 102 recv payload to 101`.
4. Audio from B does not reach A. FreeSWITCH continually logs `TS changed from last packet, resetting....` and `RTP RECV DTMF`.

### Expected behavior
In this case, B's telephone-event should be both sent and received using payload type 102, regardless of whether or not the B-side responds with `183 Session Progress` before `200 OK`.

### Package version or git hash
 - Tested on versions ranging from 1.10.7 to 1.10.9

### Trace logs
Find the log `2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:5854 sofia/SomeGateway/XYZ987 Set 2833 dtmf send payload to 102 recv payload to 101` in the logs below, and then work your way back to the SDP sent to and from the B-part (called XYZ987) and you will see how `Set 2833 dtmf [...] recv payload to 101` does not make sense.
```
2023-02-16 13:39:21.539881 92.10% [NOTICE] switch_channel.c:1123 New Channel sofia/SomeGateway/ABC123@X.X.X.X [p0OJ-35-zdiDyPoTV4Usdw..]
2023-02-16 13:39:21.539881 92.10% [DEBUG] switch_core_state_machine.c:581 (sofia/SomeGateway/ABC123@X.X.X.X) Running State Change CS_NEW (Cur 1 Tot 12)
2023-02-16 13:39:21.539881 92.10% [INFO] sofia.c:10465 sofia/SomeGateway/ABC123@X.X.X.X receiving invite from X.X.X.X:5001 version: 1.10.9-release git ceab6f2 2023-02-14 09:11:04Z 64bit call-id: p0OJ-35-zdiDyPoTV4Usdw..
2023-02-16 13:39:21.539881 92.10% [DEBUG] sofia.c:10559 verifying acl "domains" for ip/port X.X.X.X:0.
2023-02-16 13:39:21.539881 92.10% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/ABC123@X.X.X.X entering state [received][100]
2023-02-16 13:39:21.539881 92.10% [DEBUG] sofia.c:7503 Remote SDP:
v=0
o=- 1676554761 1676554761 IN IP4 X.X.X.X
s=-
c=IN IP4 X.X.X.X
t=0 0
m=audio 23258 RTP/AVP 101 96
a=rtpmap:101 opus/48000/2
a=rtpmap:96 telephone-event/48000
a=fmtp:96 0-15
a=maxptime:20
a=rtcp:23259
a=ptime:20

[routing logs...]

EXECUTE [depth=0] sofia/SomeGateway/ABC123@X.X.X.X bridge(sofia/gateway/SomeGateway/XYZ987)
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_ivr_originate.c:2291 Parsing global variables
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_ivr_originate.c:2853 Parsing session specific variables
2023-02-16 13:39:22.159877 92.10% [NOTICE] switch_channel.c:1123 New Channel sofia/SomeGateway/XYZ987 [95780dbc-9f9a-4b44-b6d0-83696e713844]
2023-02-16 13:39:22.159877 92.10% [DEBUG] mod_sofia.c:5123 (sofia/SomeGateway/XYZ987) State Change CS_NEW -> CS_INIT
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_ivr_originate.c:3161 sofia/SomeGateway/XYZ987 Setting leg timeout to 30
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_core_state_machine.c:581 (sofia/SomeGateway/XYZ987) Running State Change CS_INIT (Cur 2 Tot 13)
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_core_state_machine.c:624 (sofia/SomeGateway/XYZ987) State INIT
2023-02-16 13:39:22.159877 92.10% [DEBUG] mod_sofia.c:97 sofia/SomeGateway/XYZ987 SOFIA INIT
2023-02-16 13:39:22.159877 92.10% [INFO] sofia_glue.c:1660 sofia/SomeGateway/XYZ987 sending invite call-id: 95780dbc-9f9a-4b44-b6d0-83696e713844
2023-02-16 13:39:22.159877 92.10% [DEBUG] sofia_glue.c:1663 sofia/SomeGateway/XYZ987 sending invite version: 1.10.9-release git ceab6f2 2023-02-14 09:11:04Z 64bit
Local SDP:
v=0
o=FreeSWITCH 1676526504 1676526505 IN IP4 X.X.X.X
s=FreeSWITCH
c=IN IP4 X.X.X.X
t=0 0
m=audio 28258 RTP/AVP 101 102
a=rtpmap:101 opus/48000/2
a=fmtp:101 useinbandfec=1; maxaveragebitrate=32000; maxplaybackrate=16000; sprop-maxcapturerate=16000; ptime=20; minptime=10; maxptime=40
a=rtpmap:102 telephone-event/48000
a=fmtp:102 0-15
a=ptime:20
a=sendrecv

2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_core_state_machine.c:40 sofia/SomeGateway/XYZ987 Standard INIT
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_core_state_machine.c:48 (sofia/SomeGateway/XYZ987) State Change CS_INIT -> CS_ROUTING
2023-02-16 13:39:22.159877 92.10% [DEBUG] switch_core_state_machine.c:624 (sofia/SomeGateway/XYZ987) State INIT going to sleep
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_core_state_machine.c:581 (sofia/SomeGateway/XYZ987) Running State Change CS_ROUTING (Cur 2 Tot 13)
2023-02-16 13:39:22.179884 92.10% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/XYZ987 entering state [calling][0]
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_core_state_machine.c:640 (sofia/SomeGateway/XYZ987) State ROUTING
2023-02-16 13:39:22.179884 92.10% [DEBUG] mod_sofia.c:158 sofia/SomeGateway/XYZ987 SOFIA ROUTING
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_ivr_originate.c:67 (sofia/SomeGateway/XYZ987) State Change CS_ROUTING -> CS_CONSUME_MEDIA
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_core_state_machine.c:640 (sofia/SomeGateway/XYZ987) State ROUTING going to sleep
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_core_state_machine.c:581 (sofia/SomeGateway/XYZ987) Running State Change CS_CONSUME_MEDIA (Cur 2 Tot 13)
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_core_state_machine.c:659 (sofia/SomeGateway/XYZ987) State CONSUME_MEDIA
2023-02-16 13:39:22.179884 92.10% [DEBUG] switch_core_state_machine.c:659 (sofia/SomeGateway/XYZ987) State CONSUME_MEDIA going to sleep
2023-02-16 13:39:22.839898 91.47% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/XYZ987 entering state [proceeding][180]
2023-02-16 13:39:22.839898 91.47% [NOTICE] sofia.c:7604 Ring-Ready sofia/SomeGateway/XYZ987!
2023-02-16 13:39:22.839898 91.47% [DEBUG] switch_channel.c:3494 (sofia/SomeGateway/XYZ987) Callstate Change DOWN -> RINGING
2023-02-16 13:39:22.859879 91.47% [NOTICE] switch_ivr_originate.c:565 Ring Ready sofia/SomeGateway/ABC123@X.X.X.X!
2023-02-16 13:39:29.079885 91.33% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/XYZ987 entering state [completing][200]
2023-02-16 13:39:29.079885 91.33% [DEBUG] sofia.c:7503 Remote SDP:
v=0
o=- 1676554769 1676554769 IN IP4 X.X.X.X
s=-
c=IN IP4 X.X.X.X
t=0 0
m=audio 23298 RTP/AVP 101 102
b=RS:625
b=RR:1875
b=AS:93
a=rtpmap:101 opus/48000/2
a=fmtp:101 useinbandfec=1; maxaveragebitrate=32000; maxplaybackrate=16000; sprop-maxcapturerate=16000; ptime=20; minptime=10; maxptime=40
a=rtpmap:102 telephone-event/48000
a=fmtp:102 0-15
a=maxptime:40
a=rtcp:23299
a=ptime:20

2023-02-16 13:39:29.079885 91.33% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/XYZ987 entering state [ready][200]
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:5531 Audio Codec Compare [opus:101:48000:20:0:1]/[opus:116:48000:20:0:1]
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:5586 Audio Codec Compare [opus:116:48000:20:0:1] ++++ is saved as a match
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:5447 Set telephone-event payload to 102@48000
2023-02-16 13:39:29.079885 91.33% [DEBUG] mod_opus.c:614 Opus encoder: set bitrate based on maxaveragebitrate value found in SDP or local config [32000bps]
2023-02-16 13:39:29.079885 91.33% [DEBUG] mod_opus.c:630 Opus encoder: set audio bandwidth to [WIDEBAND] based on maxplaybackrate value found in SDP or local config [16000Hz]
2023-02-16 13:39:29.079885 91.33% [DEBUG] mod_opus.c:614 Opus encoder: set bitrate based on maxaveragebitrate value found in SDP or local config [32000bps]
2023-02-16 13:39:29.079885 91.33% [DEBUG] mod_opus.c:630 Opus encoder: set audio bandwidth to [WIDEBAND] based on maxplaybackrate value found in SDP or local config [16000Hz]
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:3754 Set Codec sofia/SomeGateway/XYZ987 opus/48000 20 ms 960 samples 0 bits 1 channels
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_codec.c:111 sofia/SomeGateway/XYZ987 Original read codec set to opus:116
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:5796 Set telephone-event payload to 102@48000

[VVV This log shows the probable error VVV]
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:5854 sofia/SomeGateway/XYZ987 Set 2833 dtmf send payload to 102 recv payload to 101
[^^^ This log shows the probable error ^^^]

2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:8655 AUDIO RTP [sofia/SomeGateway/XYZ987] X.X.X.X port 28258 -> X.X.X.X port 23298 codec: 101 ms: 20
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_rtp.c:4399 Not using a timer
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:8875 Activating RTCP PORT 23299
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_rtp.c:4708 RTCP send rate is: 1000 and packet rate is: 20000 Remote Port: 23299
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_rtp.c:2534 Setting RTCP remote addr to X.X.X.X:23299 2
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:8967 sofia/SomeGateway/XYZ987 Set 2833 dtmf send payload to 102
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:8974 sofia/SomeGateway/XYZ987 Set 2833 dtmf receive payload to 101
2023-02-16 13:39:29.079885 91.33% [DEBUG] switch_core_media.c:8997 sofia/SomeGateway/XYZ987 Set rtp dtmf delay to 40
2023-02-16 13:39:29.079885 91.33% [NOTICE] sofia.c:8686 Channel [sofia/SomeGateway/XYZ987] has been answered
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_ivr_originate.c:448 Setting codec string on sofia/SomeGateway/ABC123@X.X.X.X to opus@48000h@20i
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:5531 Audio Codec Compare [opus:101:48000:20:0:2]/[opus:116:48000:20:0:1]
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:5586 Audio Codec Compare [opus:116:48000:20:0:1] ++++ is saved as a match
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:5447 Set telephone-event payload to 96@48000
2023-02-16 13:39:29.103108 91.33% [DEBUG] mod_opus.c:614 Opus encoder: set bitrate based on maxaveragebitrate value found in SDP or local config [32000bps]
2023-02-16 13:39:29.103108 91.33% [DEBUG] mod_opus.c:630 Opus encoder: set audio bandwidth to [WIDEBAND] based on maxplaybackrate value found in SDP or local config [16000Hz]
2023-02-16 13:39:29.103108 91.33% [DEBUG] mod_opus.c:614 Opus encoder: set bitrate based on maxaveragebitrate value found in SDP or local config [32000bps]
2023-02-16 13:39:29.103108 91.33% [DEBUG] mod_opus.c:630 Opus encoder: set audio bandwidth to [WIDEBAND] based on maxplaybackrate value found in SDP or local config [16000Hz]
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:3754 Set Codec sofia/SomeGateway/ABC123@X.X.X.X opus/48000 20 ms 960 samples 0 bits 1 channels
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_codec.c:111 sofia/SomeGateway/ABC123@X.X.X.X Original read codec set to opus:116
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:5796 Set telephone-event payload to 96@48000
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:5854 sofia/SomeGateway/ABC123@X.X.X.X Set 2833 dtmf send payload to 96 recv payload to 96
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:8655 AUDIO RTP [sofia/SomeGateway/ABC123@X.X.X.X] X.X.X.X port 28640 -> X.X.X.X port 23258 codec: 101 ms: 20
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_rtp.c:4399 Not using a timer
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:8875 Activating RTCP PORT 23259
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_rtp.c:4708 RTCP send rate is: 1000 and packet rate is: 20000 Remote Port: 23259
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_rtp.c:2534 Setting RTCP remote addr to X.X.X.X:23259 2
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:8967 sofia/SomeGateway/ABC123@X.X.X.X Set 2833 dtmf send payload to 96
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:8974 sofia/SomeGateway/ABC123@X.X.X.X Set 2833 dtmf receive payload to 96
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:8997 sofia/SomeGateway/ABC123@X.X.X.X Set rtp dtmf delay to 40
2023-02-16 13:39:29.103108 91.33% [NOTICE] sofia_media.c:90 Pre-Answer sofia/SomeGateway/ABC123@X.X.X.X!
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_channel.c:3570 (sofia/SomeGateway/ABC123@X.X.X.X) Callstate Change RINGING -> EARLY
2023-02-16 13:39:29.103108 91.33% [DEBUG] switch_core_media.c:8637 Audio params are unchanged for sofia/SomeGateway/ABC123@X.X.X.X.
2023-02-16 13:39:29.103108 91.33% [DEBUG] mod_sofia.c:913 Local SDP sofia/SomeGateway/ABC123@X.X.X.X:
v=0
o=FreeSWITCH 1676526129 1676526130 IN IP4 X.X.X.X
s=FreeSWITCH
c=IN IP4 X.X.X.X
t=0 0
m=audio 28640 RTP/AVP 101 96
a=rtpmap:101 opus/48000/2
a=fmtp:101 useinbandfec=1; maxaveragebitrate=32000; maxplaybackrate=16000; sprop-maxcapturerate=16000; minptime=10; maxptime=40
a=rtpmap:96 telephone-event/48000
a=fmtp:96 0-15
a=ptime:20
a=sendrecv
a=rtcp:28641 IN IP4 X.X.X.X

2023-02-16 13:39:29.103108 91.33% [NOTICE] switch_ivr_originate.c:3844 Channel [sofia/SomeGateway/ABC123@X.X.X.X] has been answered
2023-02-16 13:39:29.103108 91.33% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/ABC123@X.X.X.X entering state [completed][200]
2023-02-16 13:39:29.219974 91.33% [DEBUG] switch_channel.c:3901 (sofia/SomeGateway/XYZ987) Callstate Change RINGING -> ACTIVE
2023-02-16 13:39:29.239884 90.80% [DEBUG] sofia.c:7493 Channel sofia/SomeGateway/ABC123@X.X.X.X entering state [ready][200]
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_channel.c:3901 (sofia/SomeGateway/ABC123@X.X.X.X) Callstate Change EARLY -> ACTIVE
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_ivr_originate.c:3902 Originate Resulted in Success: [sofia/SomeGateway/XYZ987] Peer UUID: 95780dbc-9f9a-4b44-b6d0-83696e713844
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_ivr_bridge.c:1791 (sofia/SomeGateway/XYZ987) State Change CS_CONSUME_MEDIA -> CS_EXCHANGE_MEDIA
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_core_state_machine.c:581 (sofia/SomeGateway/XYZ987) Running State Change CS_EXCHANGE_MEDIA (Cur 2 Tot 13)
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_core_state_machine.c:650 (sofia/SomeGateway/XYZ987) State EXCHANGE_MEDIA
2023-02-16 13:39:29.239884 90.80% [DEBUG] mod_sofia.c:671 SOFIA EXCHANGE_MEDIA
2023-02-16 13:39:29.239884 90.80% [INFO] switch_rtp.c:6917 Auto Changing audio RTCP port from X.X.X.X:23299 to X.X.X.X:28259
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_rtp.c:2534 Setting RTCP remote addr to X.X.X.X:28259 2
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.239884 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.259895 90.80% [DEBUG] switch_rtp.c:7511 Correct audio ip/port confirmed.
2023-02-16 13:39:29.259895 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.259895 90.80% [INFO] switch_rtp.c:6917 Auto Changing audio RTCP port from X.X.X.X:23259 to X.X.X.X:28641
2023-02-16 13:39:29.259895 90.80% [DEBUG] switch_rtp.c:2534 Setting RTCP remote addr to X.X.X.X:28641 2
2023-02-16 13:39:29.279877 90.80% [DEBUG] switch_rtp.c:1775 rtcp_stats_init: audio ssrc[3233912196] base_seq[37150]
2023-02-16 13:39:29.279877 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.299877 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.319880 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.339884 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.359885 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.379885 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.399882 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.419887 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.439885 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.459890 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.479887 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.479887 90.80% [DEBUG] switch_rtp.c:1775 rtcp_stats_init: audio ssrc[3554621532] base_seq[698]
2023-02-16 13:39:29.479887 90.80% [DEBUG] switch_rtp.c:7511 Correct audio ip/port confirmed.
2023-02-16 13:39:29.499949 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.519883 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.540038 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.559887 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.579897 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.599980 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.619883 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.639960 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.659893 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.679951 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.699899 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.699899 90.80% [DEBUG] switch_rtp.c:7756 RTP RECV DTMF
2023-02-16 13:39:29.719966 90.80% [DEBUG] switch_rtp.c:7756 RTP RECV DTMF
2023-02-16 13:39:29.759891 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.779904 90.80% [DEBUG] switch_rtp.c:666 TS changed from last packet, resetting....
2023-02-16 13:39:29.779904 90.80% [DEBUG] switch_rtp.c:7756 RTP RECV DTMF

["TS changed from last packet, resetting...." is then continually logged when B-side is silent and "RTP RECV DTMF" when speaking.]
```

**Comment by olofssonanton** [comment]
This was first reported in the Signalwire Slack community, [here](https://signalwire-community.slack.com/archives/CDC8H14PN/p1676452334633169).

Ask away if anything is unclear, or take a look at that Slack chat and see if it helps.

**Comment by olofssonanton** [comment]
I have tried to deduce where `smh->mparams->recv_te` from the following line is set to payload type 101 instead of 102.
https://github.com/signalwire/freeswitch/blob/1a2033b9157b9a801b30713c30d34cae1b000961/src/switch_core_media.c#L5850-L5851

That seems to happen here during the creation of the channel.
https://github.com/signalwire/freeswitch/blob/1a2033b9157b9a801b30713c30d34cae1b000961/src/mod/endpoints/mod_sofia/sofia_glue.c#L123

Have not yet found why it is only changed to payload type 102 when a `183 Session Progress` is received.

**Comment by olofssonanton** [comment]
Adding some further information from the chat:

> We usually receive OPUS as 102 and telephone-event as 96 from the A-side. When such a call is bridged to B, the FreeSWITCH invites with OPUS as 102 and telephone-event as 101. In those cases this issue never appears.
>
> In the case which triggers this issue however, the A-side invites with OPUS as 101 and telephone-event as 96. And then we get the "faulty" combination of OPUS as 101 and telephone-event as 102 in the invite to the B-side.
>
> So it seems to me like the incoming 101 payload type for OPUS is what causes the misbehaviour.

**Comment by ShashiKumar-SignalWire** [comment]
@olofssonanton  Please share the complete FS log of the failed call with sip trace on

**Comment by olofssonanton** [comment]
@ShashiKumar-SignalWire I unfortunately don't have the ability to share much more than the previous log.

I'll be off for a few days now, but some of my colleagues at Lynes may show up to help with this issue!

**Comment by briankwest** [comment]
Please reopen when you can provide additional details.

**Comment by olofssonanton** [comment]
@briankwest I don't seem to be able to re-open this particular issue, but we have a PR up to fix this in #1979.
